[libc++] Backport segmented iterator optimization for std::for_each to C++11 (#134960)
llvm/llvm-project: libcxx/docs/ReleaseNotes/21.rst

- Updated formatting library to Unicode 16.0.0.

> - The segmented iterator optimization for ``std::for_each``, previously available only in C++23 and later, has been
>   backported to C++11 to enable consistent optimization across standards.

Reviewer: ```suggestion
- The segmented iterator optimization for ``std::for_each`` has been backported to C++11. Previously it was only available in C++23 and later.
```
